### DISCUSSION TITLE
persist files in manifest - automatically create the file?

### DISCUSSION THREAD

**Discussion by amreus** [ScoopInstaller/Scoop#4598] [Ideas]
say you have:

```
"persist": "app.ini"
```

What it make sense to have the `persist` handler automatically create the `app.ini` file if it does not exist in the app folder after the app is unzipped?

Typically it looks like this is the way it is handled:
```
"pre_install": "if (!(Test-Path \"$persist_dir\\app.ini\")) { New-Item \"$dir\\app.ini\" | Out-Null }",
```

**Comment by amreus** [comment]
I'll answer my own question - no.

One big reason is that the app may not handle a zero-length config file gracefully.

**Reply by rashil2000** [reply]
Pretty much this. Different applications have different requirements for how the config file is created, whether it can be a hardlink, what kind of encoding it has, and of course the content. So we leave it up to the pre_install step.

**Comment by r15ch13** [comment]
https://github.com/ScoopInstaller/Scoop/pull/3212
https://github.com/ScoopInstaller/Scoop/pull/3587
